### PR TITLE
purge useless npm deps before installing

### PR DESCRIPTION
**Pull request by evgeni**
requires https://github.com/theforeman/foreman/pull/9255 and is completely untested :)

### CODE REVIEW

**Comment by ehelms** [comment]
:fireworks: :man_dancing:

File: theforeman.org/pipelines/test/testKatello.groovy [hunk 1]
```diff
@@ -106,6 +106,16 @@ pipeline {
...
                             dir('foreman') {
+                                sh "python script/filter-package-json.py"
+
+                                withRVM(["bundle exec ruby script/plugin_webpack_directories.rb > plugin_webpack.json"], ruby)
+                                script {
+                                    def plugin_webpack = readJSON file: 'plugin_webpack.json'
+                                    plugin_webpack['plugins'].each { plugin, config ->
+                                        sh "python script/filter-package-json.py --package-json ${config['root']}/package.json"
+                                    }
+                                }
```
Comment (ehelms) [inline theforeman.org/pipelines/test/testKatello.groovy:117]: Do you think that this logic can be:

 a) wrapped up in a common groovy function
 b) wrapped up in a script that lives in foreman

The two use cases I am thinking of to extrapolate this awesome logic to:

 1) Using it in other plugin CI tests
 2) Using the same logic in Katello react tests that flow through GH actions

Comment (evgeni) [reply theforeman.org/pipelines/test/testKatello.groovy:117]: Yes.

Groovy is easy, the other thing a bit more complicated but still possible.

Comment (ehelms) [reply theforeman.org/pipelines/test/testKatello.groovy:117]: Another spot I was curious if any of this logic would benefit -- https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/scripts/test/test_develop.sh#L42-L45

Comment (evgeni) [reply theforeman.org/pipelines/test/testKatello.groovy:117]: Yes, that file would benefit from being Groovy 😼

Comment (evgeni) [reply theforeman.org/pipelines/test/testKatello.groovy:117]: But no, here we actually need more than the build deps (still no storybook), but I'd expect that one to break if we apply *exactly* the same logic

**Comment by ehelms** [comment]
Since this provides a significant speed up I am happy to merge and then re-factor. As the author of the change, I'll let you have final say which order you want to go in.

**Comment by evgeni** [comment]
Merge away

**Review by ehelms — APPROVED** [review]

**Pull request merged**